openvpn/service_discovery/dto/service_proposal.go

```diff
@@ -0,0 +1,78 @@
+package dto
+
+import (
+	"github.com/mysterium/node/service_discovery/dto"
+	"encoding/json"
+	"log"
+)
+
+type proposal struct {
+	Id                int              `json:"id"`
+	Format            string           `json:"format"`
+	ServiceType       string           `json:"service_type"`
+	ProviderId        string           `json:"provider_id"`
+	PaymentMethodType string           `json:"payment_method_type"`
+	ServiceDefinition *json.RawMessage `json:"service_definition"`
+	PaymentMethod     *json.RawMessage `json:"payment_method"`
+	ProviderContacts  []dto.Contact    `json:"provider_contacts"`
+}
+
+func BuildServiceProposalFromJson(jsonData []byte) dto.ServiceProposal {
+	proposal := proposal{}
+
+	if err := json.Unmarshal([]byte(jsonData), &proposal); err != nil {
+		log.Fatal(err)
+	}
+
+	ovProposal := dto.ServiceProposal{
+		Id:                proposal.Id,
+		Format:            proposal.Format,
+		ServiceType:       proposal.ServiceType,
+		ProviderId:        dto.Identity(proposal.ProviderId),
+		PaymentMethodType: proposal.PaymentMethodType,
+		ProviderContacts:  proposal.ProviderContacts,
+	}
+
+	switch proposal.ServiceType {
+	case "openvpn":
+		ovProposal.ServiceDefinition = getServiceDefinitionFromJson(proposal.ServiceDefinition)
+	}
+
+	ovProposal.PaymentMethod = getPaymentMethodFromJson(proposal.PaymentMethodType, proposal.PaymentMethod)
+
+	return ovProposal
+}
+
+func getServiceDefinitionFromJson(rawDefinition *json.RawMessage) ServiceDefinition {
+	definition := ServiceDefinition{}
+
+	if err := json.Unmarshal(*rawDefinition, &definition); err != nil {
+		log.Fatal(err)
```

Comment:
You dont know how to handle this error in this low layer, so error should be pushed to upper layer:
```
func getServiceDefinitionFromJson() (ServiceDefinition, error) {
    if err := json.Unmarshal(*rawDefinition, &definition); err != nil {
	return nil, err
   }
}
```